xgi/core/hypergraph.py

```diff
@@ -1462,49 +1462,38 @@ def cleanup(
 
         """
         if in_place:
-            if not multiedges:
-                self.merge_duplicate_edges()
-            if not singletons:
-                self.remove_edges_from(self.edges.singletons())
-            if not isolates:
-                self.remove_nodes_from(self.nodes.isolates())
-            if connected:
-                from ..algorithms import largest_connected_component
-
-                self.remove_nodes_from(self.nodes - largest_connected_component(self))
-            if relabel:
-                from ..utils import convert_labels_to_integers
-
-                temp = convert_labels_to_integers(self).copy()
-
-                nn = temp.nodes
-                ee = temp.edges
-
-                self.clear()
-                self.add_nodes_from((n, deepcopy(attr)) for n, attr in nn.items())
-                self.add_edges_from(
-                    (e, id, deepcopy(temp.edges[id]))
-                    for id, e in ee.members(dtype=dict).items()
-                )
-                self._net_attr = deepcopy(temp._net_attr)
+            _H = self
         else:
-            H = self.copy()
-            if not multiedges:
-                H.merge_duplicate_edges()
-            if not singletons:
-                H.remove_edges_from(H.edges.singletons())
-            if not isolates:
-                H.remove_nodes_from(H.nodes.isolates())
-            if connected:
-                from ..algorithms import largest_connected_component
-
-                H.remove_nodes_from(H.nodes - largest_connected_component(H))
-            if relabel:
-                from ..utils import convert_labels_to_integers
-
-                H = convert_labels_to_integers(H)
```

Comment:
Was that wrong? Since we now use the longer relabeling version in both cases

Reply:
This is fixed by moving all of the logic to the `convert_labels_to_integers` function.

Reply:
So this becomes a one-liner again.